Sources/SwiftDocC/Semantics/DirectiveInfrastructure/DirectiveArgumentWrapper.swift

```diff
@@ -0,0 +1,181 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+protocol _DirectiveArgumentProtocol {
+    var typeDisplayName: String { get }
+    var storedAsOptional: Bool { get }
+    var required: Bool { get }
+    var name: _DirectiveArgumentName { get }
+    var allowedValues: [String]? { get }
+    
+    var parseArgument: (_ bundle: DocumentationBundle, _ argumentValue: String) -> (Any?) { get }
+    
+    func setProperty<T>(
+        on containingDirective: T,
+        named propertyName: String,
+        to any: Any
+    ) where T: AutomaticDirectiveConvertible
+}
+
+enum _DirectiveArgumentName {
+    case unnamed
+    case custom(String)
+    case inferredFromPropertyName
+}
+
+@propertyWrapper
+public struct DirectiveArgumentWrapped<Value>: _DirectiveArgumentProtocol {
```

Comment:
Does this API and the ones below need to be public? If so, we're missing documentation for them.

Reply:
They unfortunately do because you can't attach an internal property wrapper to a public property. This would actually be a good use case for the no documentation attribute @QuietMisdreavus is working on!

Curious on your thoughts here- I don't really think these should be considered truly public API but it also makes them much less user-friendly in the library itself if they're all underscored. Any suggestions? I suppose documenting them as-such is a reasonable option. 😃

Reply:
Hmm I see, so you want the projected value of the property wrapper to be public, but not the property itself? It feels like we should just document what this API does for clients, with a note that we're not expecting them to use it.